pkgs/by-name/ss/ssm-session-manager-plugin/package.nix

```diff
@@ -60,6 +60,12 @@ buildGoModule rec {
     fi
   '';
 
+  installPhase = ''
+    runHook preInstall
+    install -Dm555 $GOPATH/bin/sessionmanagerplugin-main "$out/bin/session-manager-plugin"
```

Comment:
```suggestion
    ln -s sessionmanagerplugin-main $out/bin/session-manager-plugin
```

Reply:
I noticed it because the aws cli is not recognizing the plugin as installed anymore. I've then checked the `session-manager-plugin` repository and in that makefile the binary is being build using `go build` with the `-o` parameter to define the binary name as `session-manager-plugin`:

https://github.com/aws/session-manager-plugin/blob/mainline/makefile#L87

If there is no `-o` parameter defined `go build` is using the directory as default, which is why the binary is being build as `sessionmanagerplugin-main` with this nixpkg.

I was looking for a possibility to add the `-o` parameter in this package as well but have not found one, which is why I copied how it was when it was still using `buildGoPackage` and have just changed the directory to use the `$GOPATH` variable.

I will add the suggested change and update the PR now!

Thanks a lot! :)

Reply:
I'm not too familiar with go packaging, so I don't know whether there's a proper solution but I'd expect you to be able to use the Makefile somehow, sidestepping all of these issues. The makefile could also be broken for our purposes though. 

If the binary name is entirely wrong, I'd prefer if we just renamed the installed binary. Also please add a comment briefly describing why this workaround needs to exist so that future maintainers who do know how to solve this can easily do it.